lib/datadog/grok/src/parse_grok_rules.rs

```diff
@@ -211,26 +218,40 @@ fn parse_grok_rule(
     // collect all filters to apply later
     for pattern in &grok_patterns {
         if let GrokPattern {
-            destination:
-                Some(Destination {
-                    filter_fn: Some(ref filter),
-                    ..
-                }),
+            destination: Some(destination),
             ..
         } = pattern
         {
-            let dest = pattern.destination.as_ref().unwrap();
-            let filter = GrokFilter::try_from(filter)?;
-            filters
-                .entry(dest.path.clone())
-                .and_modify(|v| v.push(filter.clone()))
-                .or_insert_with(|| vec![filter.clone()]);
+            match &destination {
+                Destination {
+                    filter_fn: Some(filter),
+                    path,
+                } => {
+                    let filter = GrokFilter::try_from(filter)?;
+                    fields
+                        .entry(destination.to_grok_field_name())
+                        .and_modify(|v| v.filters.push(filter.clone()))
+                        .or_insert_with(|| GrokField {
+                            lookup: path.clone(),
+                            filters: vec![filter.clone()],
+                        });
+                }
+                Destination {
+                    filter_fn: None,
+                    path,
+                } => {
+                    fields.entry(path.to_string()).or_insert_with(|| GrokField {
+                        lookup: path.clone(),
+                        filters: vec![],
+                    });
+                }
+            }
```

Comment:
So this seems to have been added. Is this changing the behaviour? Before if there was no filter_fn it would not get added to the rule, but now it does?

Reply:
Right, so before we'd keep only fields with filters, but now we want to keep every field to store their lookups, even if they have no filters. It's not changing the behaviour, but required for this optimization.